examples/simple.rs

```diff
@@ -1,18 +1,18 @@
-use bevy::prelude::*;
+use bevy::{prelude::*, window::PrimaryWindow};
 use bevy_egui::{egui, EguiContext, EguiPlugin};
 
 fn main() {
     App::new()
         .add_plugins(DefaultPlugins)
         .add_plugin(EguiPlugin)
-        // Systems that create Egui widgets should be run during the `CoreStage::Update` stage,
-        // or after the `EguiSystem::BeginFrame` system (which belongs to the `CoreStage::PreUpdate` stage).
+        // Systems that create Egui widgets should be run during the `CoreSet::Update` set,
+        // or after the `EguiSet::BeginFrame` system (which belongs to the `CoreSet::PreUpdate` set).
         .add_system(ui_example_system)
         .run();
 }
 
-fn ui_example_system(mut egui_context: ResMut<EguiContext>) {
-    egui::Window::new("Hello").show(egui_context.ctx_mut(), |ui| {
+fn ui_example_system(mut egui_ctx: Query<&mut EguiContext, With<PrimaryWindow>>) {
+    egui::Window::new("Hello").show(egui_ctx.single_mut().get_mut(), |ui| {
```

Comment:
can it be simplified like this?

```suggestion
fn ui_example_system(egui_ctx: Query<&EguiContext, With<PrimaryWindow>>) {
    egui::Window::new("Hello").show(egui_ctx.single(), |ui| {
```

Reply:
Only if we implement `Deref` and `DerefMut` for `EguiContext`, giving people an interface to borrow the context immutably, which I'm hesitant to do due to the following:
https://github.com/mvlabat/bevy_egui/blob/7151c40f8564650fdfad5eaa15e72d055d0a5893/src/lib.rs#L247-L256

Reply:
I ended up introducing the `EguiContexts` system param, to combine the best of both worlds: low boilerplate and mutable borrowing by default.